### ISSUE TITLE
Prometheus port in gloo-fed is wrong (9091 -> 8080)

### ISSUE THREAD

**Issue by jmunozro** [solo-io/gloo#5528]
**Describe the bug**
Error in the chart template `gloo-fed-deployment.yaml`, the annotation `prometheus.io/port` is set on port 9091 but the metrics are available on port 8080 instead

**To Reproduce**
1. kubectl port-forward deployment/gloo-fed 9999:9091 -n gloo-system
2. curl localhost:9999/metrics <---------------fails
3. kubectl port-forward deployment/gloo-fed 9999:8080 -n gloo-system
4. curl localhost:9999/metrics <---------------works

**Expected behavior**
```
spec:
  selector:
    matchLabels:
      app: gloo-fed
  template:
    metadata:
      labels:
        app: gloo-fed
      annotations:
        prometheus.io/path: /metrics
-        prometheus.io/port: "9091"
+        prometheus.io/port: "8080" 
        prometheus.io/scrape: "true"
```

**Additional context**
- Gloo Edge 1.9

**Comment by sam-heilbron** [comment]
In other templates we rely on START_STATS_SERVER (https://github.com/solo-io/gloo/blob/master/install/helm/gloo/templates/1-gloo-deployment.yaml#L178) to conditionally start the stats server for a deployment. That leverages our go-utils code here: https://github.com/solo-io/go-utils/blob/master/stats/stats.go#L53

Perhaps we could use that to isolate stats traffic to its own port?

**Comment by jenshu** [comment]
the above is a quick fix to get things working, but let's keep this issue open so we can try to figure out why gloo-fed's metrics are being served on a non-standard port, and try to make it use the same go-utils methods as our other deployments

**Comment by jmunozro** [comment]
[gloo-fed-metrics.log](https://github.com/solo-io/gloo/files/7503158/gloo-fed-metrics.log)

Please find attached and example of the /metrics call to gloo-fed pod

**Comment by sam-heilbron** [comment]
This should be resolved by the most recent Gloo EE releases: 1.10.12, 1.11.0-beta13
